### PR TITLE
Doing hw7

### PR DESCRIPTION
**Pull request by proevgenii**

### CODE REVIEW

File: hw7/hw2.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    Explanation: Both s and t become "c".
+
+    Input: a = "a#c", t = "b"
+    Output: False
+    Explanation: s becomes "c" while t becomes "b".
+
+"""
+
+
+def backspace_compare(first: str, second: str):
```
Comment (PeppaTheC) [inline hw7/hw2.py]: change varibale names, give it more sense

File: hw7/hw2.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    s = []
+    for c in first:
+        if c == "#":
+            if s:
+                s.pop()
+        else:
+            s.append(c)
+
+    t = []
+    for c in second:
```
Comment (PeppaTheC) [inline hw7/hw2.py]: same code as above

File: hw7/hw3.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+     Return value should be "x wins!"
+
+"""
+from itertools import chain
+from typing import List
+
+import numpy as np
+
+
+def tic_tac_toe_checker(board: List[List]) -> str:
```
Comment (PeppaTheC) [inline hw7/hw3.py:43]: who wins [["-", "-", "o"], ["-", "o", "o"], ["x", "x", "x"]]?

File: hw7/hw3.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def diagonals(board):
```
Comment (PeppaTheC) [inline hw7/hw3.py]: why whis function is inside another?

**Review by PeppaTheC — CHANGES_REQUESTED** [review]

File: hw7/hw3.py (hunk 3 above)
Comment (proevgenii) [reply hw7/hw3.py:43]: my code shows 'wins x'. 
It is not right?

File: hw7/hw2.py (hunk 1 above)
Comment (proevgenii) [reply hw7/hw2.py]: Tried to fix

File: hw7/hw2.py (hunk 2 above)
Comment (proevgenii) [reply hw7/hw2.py]: Fixed, by making a outer func

File: hw7/hw3.py (hunk 4 above)
Comment (proevgenii) [reply hw7/hw3.py]: Fixed

**Review by PeppaTheC — APPROVED** [review]

**Pull request merged**